#### A: Что нас волнует?
В предыдущих главах мы работали с данными без их предварительной очистки.

В этой и следующих задачах мы используем набор данных с фильмами и сериалами компании Disney. CSV файл доступен под названием 'disney_title.csv'. Ознакомиться более подробно с датасетом можно по следующей ссылке: https://www.kaggle.com/datasets/shivamb/disney-movies-and-tv-shows.

В этой задаче выведите информацию о датасете, используя функцию .info()

In [4]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   type          1450 non-null   object
 1   title         1450 non-null   object
 2   director      977 non-null    object
 3   cast          1260 non-null   object
 4   country       1231 non-null   object
 5   release_year  1450 non-null   int64 
 6   rating        1447 non-null   object
 7   duration      1450 non-null   object
 8   listed_in     1450 non-null   object
 9   description   1450 non-null   object
 10  Date          1447 non-null   object
dtypes: int64(1), object(10)
memory usage: 124.7+ KB
None


#### B: Какая сегодня дата?
В предыдущем задании мы обнаружили, что переменная Date из датасета по фильмам и сериалам компании Disney имеет тип object. В этом задании мы поменяем тип переменной с object на datetime с помощью функции pd.to_datetime(). В выводе распечатайте тип столбца ['Date'] с помощью .dtype

CSV файл доступен под названием 'disney_title.csv'.

In [8]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df['Date'] = pd.to_datetime(df['Date'])

print(df['Date'].dtype)

datetime64[ns]


#### C: А что выходило за год?
В предыдущем задании мы изменили тип переменной, содержащей дату, с объекта на значение даты и времени. В этом задании мы отфильтруем датасет с использованием переменной. В начале еще раз измените тип переменной Date на datetime64 и отфильтруйте все фильмы и сериалы, которые выходили с 1 января 2020 года по 1 января 2021 года. В результате выведите переменную с названиями десяти первых фильмов после фильтрации с помощью функции head().

CSV файл доступен под названием 'disney_title.csv'.

Формат вывода
284 American Dragon: Jake Long{br} 285 Car SOS{br} 286 Disney Channel Games 2008{br} 287 Disney Roll It Back (Shorts){br} 288 Dog With a Blog{br} 289 Mickey Go Local{br} 290 Myth: A Frozen Tale{br} 291 Okavango: River of Dreams{br} 292 Pair of Kings{br} 293 Secrets of Sulphur Springs{br} Name: title, dtype: object

In [39]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df['Date'] = pd.to_datetime(df['Date'])

filtered_df = df[(df['Date'] >= '2020-01-01') & (df['Date'] < '2021-01-01')]

print(filtered_df['title'].head(10))

325                                         Burrow
326                        Cosmos: Possible Worlds
327    Disney Gallery / Star Wars: The Mandalorian
328                          Max Keeble's Big Move
329                                           Soul
330                      Arendelle Castle Yule Log
331                       Buried Truth of the Maya
332                    Disney Parks Sunrise Series
333                                Dory's Reef Cam
334                                 Into the Woods
Name: title, dtype: object


#### D: Удаление
В этом задании мы продолжаем работать с CSV файлом под названием 'disney_title.csv'. Вам необходимо удалить столбец из датасета 'release_year' и вывести название оставшихся столбцов с помощью функции list().

In [ ]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df = df.drop(columns=['release_year'])

print(list(df.columns))

['type',
 'title',
 'director',
 'cast',
 'country',
 'rating',
 'duration',
 'listed_in',
 'description',
 'Date']

#### E: А можно переименовать?
Умение переименовывать датасеты помогает при работе с большими объемами данных. В этом задании вам необходимо поменять название переменных из 'disney_title.csv' на такие же названия написанные с большой буквы. В результате выведите названия стобцов с помощью функции list().

In [46]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df.columns = df.columns.str.capitalize()

print(list(df.columns))

['Type', 'Title', 'Director', 'Cast', 'Country', 'Release_year', 'Rating', 'Duration', 'Listed_in', 'Description', 'Date']


#### F: А это что за покемон?
В этом задании мы продолжаем работать с CSV файлом доступным под названием 'disney_title.csv'.

В этом задании вам необходимо заменить "&" на "," в столбце "listed_in", сохранить результат в новую переменную, вывести последние пять значений в новой переменной с помощью функции 'tail()'.

In [64]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df['listed_in1'] = df['listed_in'].str.replace('&', ',')

print(df['listed_in1'].tail())

1445        Action-Adventure, Family, Science Fiction
1446                 Action-Adventure, Comedy, Family
1447                      Biographical, Comedy, Drama
1448                     Buddy, Comedy, Coming of Age
1449    Action-Adventure, Animals , Nature, Animation
Name: listed_in1, dtype: object


#### G: Сколько пропусков?
CSV-файл доступен под названием 'disney_title.csv'.

В этой задаче требуется вывести количество пропущенных значений по каждому стобцу.

In [69]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

print(df.isnull().sum())

type              0
title             0
director        473
cast            190
country         219
release_year      0
rating            3
duration          0
listed_in         0
description       0
Date              3
dtype: int64


#### H: Удаление?
CSV-файл доступен под названием 'disney_title.csv'. По результатам седьмого задания мы выяснили, что небольшое количество пропущенных значений (по три) находятся в столбцах 'rating' и 'Date'. Воспользуйтесь функцией 'dropna()' для удаления пропущенных значений во всех столбцах. В результате выведите количество пропущенных переменных по каждому столбцу в вашем датасете.

In [84]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df = df.dropna()

print(df.isnull().sum())

type            0
title           0
director        0
cast            0
country         0
release_year    0
rating          0
duration        0
listed_in       0
description     0
Date            0
dtype: int64


#### I: Какой процент пропусков?
В предыдущем задании мы смотрели на пропущенные значения без анализа их количества относительно количества заполненных значений во всем датасете. В этом задании мы продолжим работать с CSV файлом под названием 'disney_title.csv'. Выведите процент пропущенных значений относительно всех значений. Результат округлите до сотых с помощью функции "round()".

In [85]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

print(round(df.isnull().mean()*100, 2))

type             0.00
title            0.00
director        32.62
cast            13.10
country         15.10
release_year     0.00
rating           0.21
duration         0.00
listed_in        0.00
description      0.00
Date             0.21
dtype: float64


#### J: Замена пропусков
Из предыдущего задания мы выяснили, что наибольшие проценты пропущенных значений в CSV файле 'disney_title.csv' находятся в стобцах director (32.62%), cast (13.10%) и country (15.10%). В этом задании нужно заменить пропущенные значения в столбце 'country' на 'Country not specified' с помощью функции 'fillna'. Выведите первые пять первых значений переменной 'country'.

In [90]:
import pandas as pd

df = pd.read_csv('disney_title.csv')

df['country'] = df['country'].fillna('Country not specified')

print(df['country'].head())

0    Country not specified
1    Country not specified
2            United States
3    Country not specified
4    Country not specified
Name: country, dtype: object
